### PR TITLE
crew: Allow to use external prebuilt binary (like .deb and .rpm)

### PR DESCRIPTION
**Pull request by supechicken**
Package need to execute `is_external` to use external binary

Tested on `x86_64`

### CODE REVIEW

File: crew [hunk 1]
```diff
@@ -495,11 +495,12 @@ end
 
 def download
   url = @pkg.get_url(@device[:architecture])
+  @external = @pkg.is_external?
```
Comment (uberhacker) [inline crew:498]: If you want to use a property, you will need to add it to `lib/package.rb` also.  However, if you do this then every package needs this property set to a default value if not set directly.

**Comment by uberhacker** [comment]
I'm beginning to regret my decision to allow deb package support in crew. :)

File: crew (hunk 1 above)
Comment (supechicken) [reply crew:498]: Like `is_fake`, the new `is_external` function in `package.rb` will return `nil` if the variable is not set

Comment (supechicken) [reply crew:498]: (which equal to `false`)

**Comment by uberhacker** [comment]
Maybe I'm off base here but doesn't this issue go away if we use source_url instead of binary_url for `.deb` archives?

**Comment by supechicken** [comment]
> Maybe I'm off base here but doesn't this issue go away if we use source_url instead of binary_url for `.deb` archives?

I think the package archives like `deb` are more like `prebuilt binary` instead of a `source` (although they are not built and packaged by us) The another reason that using `binary_url` is to skip the `rename *.la to *.la_tmp` part and won't show the `gcc` version (not necessary, but looks better:) )

**Comment by uberhacker** [comment]
> > Maybe I'm off base here but doesn't this issue go away if we use source_url instead of binary_url for `.deb` archives?
> 
> I think the package archives like `deb` are more like `prebuilt binary` instead of a `source` (although they are not built and packaged by us) The another reason that using `binary_url` is to skip the `rename *.la to *.la_tmp` part and won't show the `gcc` version (not necessary, but looks better:) )

I would agree with that statement if deb packages were native to ChromeOS.  However, since they need to be reconfigured to work in Chromebrew, I tend to think they qualify as source instead of binary.

**Comment by supechicken** [comment]
Close due to `source_url` is better

**Pull request closed without merging**